akka-http-core/src/main/scala/akka/http/engine/client/ConnectionPool.scala

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.client
+
+import java.net.InetSocketAddress
+import scala.util.{ Success, Try }
+import scala.concurrent.{ Promise, Future }
+import akka.event.LoggingAdapter
+import akka.actor.{ ActorSystem, Props, ActorRef }
+import akka.stream.{ FlattenStrategy, FanOutShape2, FlowShape, FlowMaterializer }
+import akka.stream.impl.ActorProcessor
+import akka.stream.scaladsl._
+import akka.http.model._
+import akka.http.util._
+import akka.http.Http
+
+private[http] object ConnectionPool {
+  import HostConnectionPoolGateway._
+
+  type ClientFlow = Flow[(HttpRequest, Any), (Try[HttpResponse], Any), Http.HostConnectionPool]
+
+  final class HostPool(flow: ClientFlow, val isShutdown: Future[Unit], val doShutdown: () ⇒ Future[Unit]) {
+    def clientFlow[T] = flow.asInstanceOf[Flow[(HttpRequest, T), (Try[HttpResponse], T), Http.HostConnectionPool]]
+  }
+
+  val CowboyEntry = new HostPool(null, null, null) // for reigning in potentially approaching thundering herds
```

Comment:
And now you need to explain why "Sleep! Sleep!" or "Park! Park!" is better than "Yield! Yield!"... :)